includes/components/class-admin.php

```diff
@@ -1225,7 +1225,7 @@ public function render_meta_box( $post, $args ) {
 						}
 
 						// Set field value.
-						if ( '' !== $value ) {
+						if ( '' !== $value || ( 'checkbox' === $field_args['type'] && ! in_array( $post->post_status, [ 'auto-draft', 'draft' ], true ) && hp\get_array_value($field_args, 'default', null) !== $value ) ) {
```

Comment:
Let me know if this issue occurs with checkboxes only, or also with e.g. text fields

Reply:
As far as I've seen, the problem only appears with checkboxes

Reply:
Please test also select, radio and checkboxes (not checkbox) field type or you have done it already?

Reply:
Let me know what's the result, Select fields are also ok?

Reply:
I've tested it with select, multiple select, radio, checkboxes, text, number and the bug is reproducing as well, I suppose this problem persist with all of the types

Reply:
Please make fix for all fields in this case. It seems like that removing condition for checking checkbox is enough to solve this issue